sds_in_a_box/sds_in_a_box_stack.py

```diff
@@ -72,27 +83,122 @@ def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+                                                     user_pool_client_name= f"sdscommandline-{SDS_ID}",
+                                                     id_token_validity=cdk.Duration.minutes(60),
+                                                     access_token_validity=cdk.Duration.minutes(60),
+                                                     refresh_token_validity=cdk.Duration.minutes(60),
+                                                     auth_flows=cognito.AuthFlow(admin_user_password=True,
+                                                                                 user_password=True,
+                                                                                 user_srp=True,
+                                                                                 custom=True),
+                                                     prevent_user_existence_errors=True)
+        
+        # Add a random unique domain name where users can sign up / reset passwords
+        # Users will be able to reset their passwords at https://sds-login-{SDS_ID}.auth.us-west-2.amazoncognito.com/login?client_id={}&redirect_uri=https://example.com&response_type=code
+        userpooldomain = userpool.add_domain(id="TeamLoginCognitoDomain",
+                                             cognito_domain=cognito.CognitoDomainOptions(domain_prefix=f"sds-login-{SDS_ID}"))
+
+        # Add a lambda function that will trigger whenever an email is sent to the user (see the lambda section above)
+        userpool.add_trigger(cognito.UserPoolOperation.CUSTOM_MESSAGE, signup_lambda)
+
+        # Create an initial user of the API
+        initial_user = cognito.CfnUserPoolUser(self, "MyCfnUserPoolUser",
+                                               user_pool_id=userpool.user_pool_id,
+                                               desired_delivery_mediums=["EMAIL"],
+                                               force_alias_creation=False,
+                                               user_attributes=[cognito.CfnUserPoolUser.AttributeTypeProperty(
+                                                  name="email",
+                                                  value="harter@lasp.colorado.edu"
+                                               )],
+                                               username="harter@lasp.colorado.edu"
+                                              )
+
+########### OUTPUTS
+        # This is a list of the major outputs of the stack
+        cdk.CfnOutput(self, "API_URL", value=api_url.url)
+        cdk.CfnOutput(self, "COGNITO_USERPOOL_ID", value=userpool.user_pool_id)
+        cdk.CfnOutput(self, "COGNITO_APP_ID", value=command_line_client.user_pool_client_id)
+        cdk.CfnOutput(self, "SIGN_IN_WEBPAGE", value=f"https://sds-login-{SDS_ID}.auth.us-west-2.amazoncognito.com/login?client_id={command_line_client.user_pool_client_id}&redirect_uri=https://example.com&response_type=code")
```

Comment:
Do we have an imap url yet? Might be nice to use `dev.imap-soc.com` `imap-soc.com` for nicer domains?

Reply:
Nope no URL yet, we should get started thinking about that